movies.js

```diff
@@ -0,0 +1,69 @@
+/* eslint no-restricted-globals: 'off' */
+
+// Iteration 1: Ordering by year - Order by year, ascending (in growing order)
+
+function orderByYear(arr) {
+  const sorted = [...arr].sort(function(a, b) {
+    if (a.year === b.year) {
+      return a.title.localeCompare(b.title);
+    }
+    return a.year - b.year;
+  });
+  return sorted;
+}
+
+// Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct
+
+function howManyMovies(array) {
+  let numberOfDrama = 0;
+  const dramaMovies = array.filter(function(drama) {
+    if (
+      drama.genre.includes("Drama") &&
+      drama.director === "Steven Spielberg"
+    ) {
+      return true;
+    }
+  });
+  numberOfDrama = dramaMovies.length;
+  return numberOfDrama;
+}
+
+// Iteration 3: Alphabetic Order - Order by title and print the first 20 titles
+
+function orderAlphabetically(array){
+let newArray = array.map(function(order){
+    return order.title
+});
+let title = newArray.sort();
+return title.slice(0,20);
+}
+
+// Iteration 4: All rates average - Get the average of all rates with 2 decimals
+
+function ratesAverage(array){
+const rating = array.reduce(accumulator, currentValue){
+    return currentValue.rate / array.length 
+}
+}
```

Comment:
Nice start, it's a good idea to use the reduce() method here.
Watch out the syntax though, (accumulator, currentValue) are the 2 arguments of the callback function inside reduce(). You could do something like this:

```
function ratesAverage(array) {
  if (array.length === 0) return 0;
  const rating = array.reduce((accumulator, currentValue) => {
    if(currentValue.rate) return accumulator + currentValue.rate;
    else return accumulator;
    }, 0) / array.length;
  return parseFloat(rating.toFixed(2));
}
```